sdk/identity/src/token_credentials/default_credentials.rs

```diff
@@ -67,8 +66,11 @@ pub enum DefaultCredentialError {
     EnvironmentCredentialError(#[from] super::EnvironmentCredentialError),
     #[error("Error getting managed identity credential: {0}")]
     ManagedIdentityCredentialError(#[from] super::ManagedIdentityCredentialError),
-    #[error("End of default list")]
-    EndOfDefaultList,
+    #[error(
+        "Multiple errors were encountered while attempting to authenticate:{}",
```

Comment:
Nit: I think we should have a `\n` after the ":" (see below where I've suggested a change to `format_aggregate_error`